src/main/kotlin/three/consulting/epoc/service/CustomerService.kt

```diff
@@ -24,5 +24,16 @@ class CustomerService(private val customerRepository: CustomerRepository) {
         }
         else throw UnableToCreateCustomerException()
     }
+
+    fun updateCustomerForId(id: Long, customerRequest: CustomerDTO): CustomerDTO {
+        val existingCustomer = customerRepository.findByIdOrNull(id)
+        if (existingCustomer != null) {
+            val customer = Customer(customerRequest)
+            return CustomerDTO(customerRepository.save(customer))
+        }
+        else throw UnableToUpdateCustomerException()
+    }
```

Comment:
Same here, drop the id function parameter and check if the `existingCustomer.id` is null or not. 

You can also drop the double repository query and just run save. The intended behavior will be implemented in a proper way on the frontend side that creating new entities with PUT won't happen, just updates.